crypto/engine/eng_int.h

```diff
@@ -8,8 +8,8 @@
  * https://www.openssl.org/source/license.html
  */
 
-#ifndef HEADER_ENGINE_INT_H
-# define HEADER_ENGINE_INT_H
+#ifndef OPENSSL_ENGINE_INT_H
+# define OPENSSL_ENGINE_INT_H
```

Comment:
The include guard should be OPENSSL_ENG_INT_H or none at all?

Reply:
It should be renamed to something like `eng_locl.h` and not include any guards.  That might possibly be subject for a separate PR.

Reply:
> It should be renamed to something like eng_locl.h and not include any guards. That might possibly be subject for a separate PR.

I decided to refrain from backporting to 1.1.1 and do a complete header cleanup instead. So I will move or rename `eng_int.h`. Is there any reason that speaks against placing `eng_int.h` (together with `crmf_int.h`, `cmp_int.h`, `pcy_int.h`)  into `include/internal`?

```
$ find -name '*_int.h'
./crypto/crmf/crmf_int.h
./crypto/engine/eng_int.h
./crypto/include/internal/err_int.h
./crypto/include/internal/ess_int.h
./crypto/include/internal/ec_int.h
./crypto/include/internal/rand_int.h
./crypto/include/internal/store_int.h
./crypto/include/internal/asn1_int.h
./crypto/include/internal/modes_int.h
./crypto/include/internal/cryptlib_int.h
./crypto/include/internal/bn_int.h
./crypto/include/internal/evp_int.h
./crypto/include/internal/x509_int.h
./crypto/cmp/cmp_int.h
./crypto/x509/pcy_int.h
```

Having two possible endings for the header files located in the source subdirectories `*_locl.h`, `*_lcl.h`is also confusing. Which variant do you prefer?

Reply:
> The include guard should be OPENSSL_ENG_INT_H or none at all?

done

Reply:
> It should be renamed to something like eng_locl.h and not include any guards. 

That's even better